### DISCUSSION TITLE
How does one run SA currently?

### DISCUSSION THREAD

**Discussion by Hans5958** [ScratchAddons/ScratchAddons#80] [Help] [answered]
How does someone build Scratch Addons from the source code and run it currently?

Also, is this Chrome only? No Firefox?

**Comment by WorldLanguages** [comment, marked as answer]
No need to build Scratch Addons. It works as is. Just clone the repo and import the folder (or manifest.json for Firefox) into extension debug mode.
Right now you can install it, but the settings UI is not done yet, so toggling and changing settings from the console is the only option and not something I'll take the time to explain how it works lol.
Firefox is 100% supported with slightly less functionality (for example, notification buttons aren't supported by Firefox), and Chrome is still preferred if you're developing an addon and debugging it.

**Comment by Hans5958** [comment]
Ah, okay. I am pretty new it the development world of extension. 

The only thing that is closely related to the extension development is PreMiD, and that thing works by developing presences that is seperate that the extension itself. Developers can develop their presences without touching the extension code, and users can install one of the presences that they want. (this might be an idea for #65)

Having me developing directly to the extension would be an interesting challenge to me.

**Comment by WorldLanguages** [comment]
Yeah I must admit the addon loader gets into complicated stuff, like ES6 modules, revocable proxies, using the DOM to send messages to userscripts... But that's exactly why I'm making it! We can forget about all of these and focus on the addons afterwards. And adding new features above the current system is easy.

**Comment by WorldLanguages** [comment]
Read the README if you haven't. It explains a little bit more how this works.

**Comment by Hans5958** [comment]
Another point, I think it is better to have the installation steps on the Readme, just in case there are peasants like me.

**Comment by WorldLanguages** [comment]
Yeah, I will do that soon. There will be a guide on creating and debugging addons :)

**Comment by Hans5958** [comment]
> 
> 
> Yeah I must admit the addon loader gets into complicated stuff, like ES6 modules, revocable proxies, using the DOM to send messages to userscripts... But that's exactly why I'm making it! We can forget about all of these and focus on the addons afterwards. And adding new features above the current system is easy.
> 
> Read the README if you haven't. It explains a little bit more how this works.

Yeah, I have took the time of trying to understand of how it works. Well, not much, because you still didn't done that wiki page. In the end, you know better than me. 

By the way, I think my suggestion related to PreMiD's approach in handling the case is interesting, but just take it with a grain of salt.

**Comment by WorldLanguages** [comment]
@Hans5958, I wrote some documentation for this repo's wiki. Please take a look at it and tell me what you think.
Closing, but feel free to reply here.
